### PR TITLE
Avoid deprecated g_type_class_add_private under src

### PR DESCRIPTION
**Pull request by yetist**
Changed CajaInformationPanel, CajaLocationBar and CajaNavigationAction

### CODE REVIEW

**Comment by raveit65** [comment]
Ok, this is more that a simple replacement of a deprecation.
Would be nice if this can reviewed by more people.
@mate-desktop/contributors @mate-desktop/core-team

**Comment by lukefromdc** [comment]
At least the code change for this one is not as complex as some of the others though it does remove `caja_navigation_action_get_type` I see that is exposed in caja-navigation-action.h but that a string search for "caja_navigation_action_get_type" in mate-search-tool does not find that string anywhere in the code except in caja-navigation-action.c, implying that that is unused code unless used by extensions (it's not found in our own caja-extensions package).

Thus removing that block and references to it should cause no problems unless a third party extension can use it, and the rest of the code changes in this particular PR are not so complex as some of the others.

**Comment by yetist** [comment]
The _get_type function is generated by the G_DEFINE_TYPE macro.

**Comment by lukefromdc** [comment]
OK, I can see that in the code now but I really don't know enough about these macros to catch that on my own.  Most of the rest of these PR's (other MATE apps) have been boilerplate that seemed to always work. Caja is quite complex, heck the original Nautilus was Easel's sole released product. The libunique->GtkApplication transition was also brutally complex, mostly guided by what Nautilus had done but not entirely because the 2.32>3.0 transition for Nautilus also dropped some features I had to add back. That was months of work spanning from MATE 1.15 to 1.17.

Nautilus still has this in seven files,  and seven more either converted to or used the G_DEFINE_TYPE_WITH_PRIVATE macro from the start of some other change:

nautilus_files_3 30

**Comment by yetist** [comment]
@lukefromdc 

The nautilus files in the left window you pasted can see the compile warning in the build log:
https://gitlab.gnome.org/GNOME/nautilus/-/jobs/199162

File: src/caja-navigation-action.c [hunk 1]
```diff
@@ -58,33 +54,7 @@ enum
     PROP_WINDOW
 };
 
-GType
-caja_navigation_action_get_type (void)
```
Comment (vkareh) [inline src/caja-navigation-action.c]: The prototype for this function is still in the header file, and it's used in a type macro? what does this function do anyway?

Comment (vkareh) [reply src/caja-navigation-action.c]: okay I should have read the earlier comments, but still there is a prototype for this function. If the function is generated by the macro, why do we need a prototype for it then?

**Comment by lukefromdc** [comment]
Should I go ahead and merge this or is further review needed?

**Comment by yetist** [comment]
I think there is no problem about `caja_navigation_action_get_type`:

1. The prototype of this function in the header file is required.
Maybe other files will call `caja_navigation_action_get_type` by `CAJA_TYPE_NAVIGATION_ACTION`, and I got one here: [src/caja-navigation-window-menus.c#L943](https://github.com/mate-desktop/caja/blob/master/src/caja-navigation-window-menus.c#L943)


2. `caja_navigation_action_get_type` is generated by the `G_DEFINE_TYPE_WITH_PRIVATE` macro in the body file.
This can be confirmed by the following command:
```
$ cat test.c 
#include <gtk/gtk.h>

G_DEFINE_TYPE_WITH_PRIVATE (CajaNavigationAction, caja_navigation_action, GTK_TYPE_ACTION)
$ gcc -E test.c `pkg-config --cflags gtk+-3.0` | tail | indent -
```

**Comment by yetist** [comment]
Rebase on the master done.

**Pull request merged**